src/vault.ts

```diff
@@ -0,0 +1,83 @@
+import { Batch } from '@mparticle/event-models';
+import { Dictionary, isEmptyObject } from './utils';
+
+export default class Vault {
+    // TODO: Make this generic
+    public contents: Dictionary<Batch>;
+
+    constructor(public key: string) {
+        this.key = key;
```

Comment:
Yea I did see that.  I was thinking about it being overwritten internally. Will TS be smart enough to know that it is a string if it is being set immediately by the constructor?  If so then it'd be overkill to add it.  Otherwise, we should make the key private so that it can only ever be set by the constructor.